crypto/elliptic-curve.ts

```diff
@@ -328,20 +395,31 @@ function projectiveEqual(g: GroupProjective, h: GroupProjective, p: bigint) {
   return mod(g.y * hz3, p) === mod(h.y * gz3, p);
 }
 
-function projectiveOnCurve({ x, y, z }: GroupProjective, p: bigint, b: bigint) {
+function projectiveOnCurve(
+  { x, y, z }: GroupProjective,
+  p: bigint,
+  b: bigint,
+  a: bigint
+) {
   // substitution x -> x/z^2 and y -> y/z^3 gives
-  // the equation y^2 = x^3 + b*z^6
+  // the equation y^2 = x^3 + a*z^4 + b*z^6
   // (note: we allow a restricted set of x,y for z==0; this seems fine)
   let x3 = mod(mod(x * x, p) * x, p);
   let y2 = mod(y * y, p);
-  let z3 = mod(mod(z * z, p) * z, p);
-  let z6 = mod(z3 * z3, p);
-  return mod(y2 - x3 - b * z6, p) === 0n;
+  let z2 = mod(z * z, p);
+  let z4 = mod(z2 * z2, p);
+  let z6 = mod(z4 * z2, p);
+  return mod(y2 - x3 - a * x * z4 - b * z6, p) === 0n;
```

Comment:
This last equation does not seem to align with the comment above. Meaning, it says $y^2 = x^3 + az^4 + b z^6$ but the `a` term seems to have an additional `x` here `y2 - x3 - a * x * z4 - b * z6`

Reply:
good catch! the comment is wrong :)